source/dem/particle_particle_heat_transfer.cc

```diff
@@ -0,0 +1,225 @@
+  // with microscale numerical modeling of thermofluid and pebble-pebble
+  // interactions (Doctoral dissertation, University of California, Los Angeles)
+  // Equations 3.25 to 3.29.
+}
+
+double
+calculate_solid_macrogap_resistance(const double radius_one,
+                                    const double radius_two,
+                                    const double thermal_conductivity_one,
+                                    const double thermal_conductivity_two,
+                                    const double contact_radius)
+{
+  // resistance_i = characteristic length parallel to heat flux /
+  // (thermal_conductivity
+  // * characteristic area perpendicular to the heat flux)
+  const double resistance_one =
+    (M_PI * radius_one / 4.0) * 1 /
+    (M_PI * (radius_one * radius_one - contact_radius * contact_radius) *
+     thermal_conductivity_one);
+  const double resistance_two =
+    (M_PI * radius_two / 4.0) * 1 /
+    (M_PI * (radius_two * radius_two - contact_radius * contact_radius) *
+     thermal_conductivity_two);
+
+  return resistance_one + resistance_two;
+}
+
+double
+calculate_interstitial_gas_microgap_resistance(
+  const double equivalent_surface_roughness,
+  const double contact_radius,
+  const double gas_parameter_m,
+  const double thermal_conductivity_gas,
+  const double maximum_pressure,
+  const double effective_microhardness)
+{
+  const double x_1 = 2.0 * maximum_pressure / effective_microhardness;
+  const double x_2 = 0.03 * maximum_pressure / effective_microhardness;
+
+  // If the values are out of the valid range ]0, 2[ of the erfc^-1 function,
+  // this resistance has no physical meaning. Particles are too close or not
+  // close enough for a microgap between them to make sense. This resistance is
+  // used in parallel with another one, so putting it to infinity ignores it in
+  // the total resistance.
+  if (x_1 >= 2.0 || x_1 <= 0.0)
+    {
+      return INFINITY;
+    }
+
+  const double a_1 = boost::math::erfc_inv(x_1);
+  const double a_2 = boost::math::erfc_inv(x_2) - a_1;
+
+  return (2 * std::sqrt(2) * equivalent_surface_roughness * a_2) /
+         (M_PI * thermal_conductivity_gas * contact_radius * contact_radius *
+          std::log(1 + a_2 / (a_1 + gas_parameter_m /
+                                      (2 * std::sqrt(2) *
+                                       equivalent_surface_roughness))));
+}
+
+double
+calculate_interstitial_gas_macrogap_resistance(
+  const double harmonic_radius,
+  const double thermal_conductivity_gas,
+  const double contact_radius,
+  const double gas_parameter_m)
+{
+  const double A = 2 * sqrt(harmonic_radius * harmonic_radius -
+                            contact_radius * contact_radius);
+  const double S = 2 * (harmonic_radius - (contact_radius * contact_radius) /
+                                            (2 * harmonic_radius)) +
+                   gas_parameter_m;
```

Comment:
```suggestion
  const double A = 2. * sqrt(harmonic_radius * harmonic_radius -
                            contact_radius * contact_radius);
  const double S = 2. * (harmonic_radius - (contact_radius * contact_radius) /
                                            (2. * harmonic_radius)) +
                   gas_parameter_m;
```

If it's a double, please put the dot.